core-bundle/src/Resources/contao/modules/ModuleSearch.php

```diff
@@ -313,13 +315,61 @@ protected function compile()
 					$objTemplate->hasContext = true;
 				}
 
+				$this->addImageToTemplateFromSearchResult($arrResult[$i], $objTemplate);
+
 				$this->Template->results .= $objTemplate->parse();
 			}
 
 			$this->Template->header = vsprintf($GLOBALS['TL_LANG']['MSC']['sResults'], array($from, $to, $count, $strKeywords));
 			$this->Template->duration = System::getFormattedNumber($query_endtime - $query_starttime, 3) . ' ' . $GLOBALS['TL_LANG']['MSC']['seconds'];
 		}
 	}
+
+	protected function addImageToTemplateFromSearchResult(array $result, Template $template): void
+	{
+		$template->hasImage = false;
+
+		if (!isset($result['meta']) || null == $result['meta'])
```

Comment:
This does not make sense. If `$result['meta']` is `null`, the `!isset()` check would already match. Did you mean `!$result['meta']` instead? In case it is an empty string or false?

Reply:
But it’s only a `==` not `===` so it would also match `false`, `0` and more...

Reply:
Improved in 9d8bd2be28ec1c46475512b2ba94e73c0f09803c.

Reply:
I still do not get it. Isn't `if (!array_key_exists('meta', $result) || null === $result['meta'])` the same as `if (!isset($result['meta']))`?

Reply:
Yes, it is: https://3v4l.org/GDsMU